### PR TITLE
Add validator with line number and default setter

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: jsonschema_gentypes/validate.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+"""
+Package that ofer some utilities function to validate the data against a JSON schema.
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:2]:
```suggestion
Module that offer some useful functions to validate the data against a JSON schema.
```

File: jsonschema_gentypes/validate.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        validator_class,
+        {"properties": set_defaults},
+    )
+
+
+def validate(
+    filename: str, data: Dict[str, Any], schema: Dict[str, Any], default: bool = True
+) -> Tuple[List[str], Dict[str, Any]]:
+    """
+    Vaildate the YAML, with him JSON schema.
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:80]:
```suggestion
    Validate the YAML, with it's JSON schema.
```

File: jsonschema_gentypes/validate.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    Arguments:
+        filename: Name used to generate the error message
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:83]:
```suggestion
        filename: Name used to generate the error messages
```

File: jsonschema_gentypes/validate.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        data: The data structure to be validated (should be loaded with ruamel.yaml to have the lines number)
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:84]:
```suggestion
        data: The data structure to be validated (should be loaded with ruamel.yaml to have the lines numbers)
```

File: jsonschema_gentypes/validate.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        schema: The loaded JSON schema
+        default: Should fill the data with the default provided in the JSON schema
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:86]:
```suggestion
        default: If true, fill the data with the defaults provided in the JSON schema
```

File: jsonschema_gentypes/validate.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+
+def load(
+    filename: str,
+    schema: Dict[str, Any],
+    default: bool = True,
+    exit_on_error: bool = True,
+) -> Any:
+    """
+    Load and vaildate the YAML, with him JSON schema.
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:139]:
```suggestion
    Load and validate the YAML, with it's JSON schema.
```

File: jsonschema_gentypes/validate.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+    Arguments:
+        filename: Name of the file that should be loaded
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:142]:
```suggestion
        filename: Path of the file that should be loaded
```

File: jsonschema_gentypes/validate.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        default: Should fill the data with the default provided in the JSON schema
+        exit_on_error: Should exit on error
+    """
+    with open(filename) as data_file:
+        errors, data = validate(filename, ruamel.yaml.round_trip_load(data_file), schema, default)  # type: ignore
+
+    if errors:
+        LOG.error("The config file is invalid:\n%s", "\n".join(errors))
+        if exit_on_error:
+            sys.exit(1)
```
Comment (arnaud-morvan) [inline jsonschema_gentypes/validate.py:153]: I would prefer a raise here (API not CLI). Caller could handle it as he wants.

**Review by arnaud-morvan — APPROVED** [review]
Minor comments and typos.

File: jsonschema_gentypes/validate.py (hunk 8 above)
Comment (sbrunner) [reply jsonschema_gentypes/validate.py]: Then I also remove the exit_on_error...

**Pull request merged**